### PR TITLE
update libsonic to v0.5.1

### PR DESCRIPTION
**Pull request by gordielachance**
makes plugin work again on Kodi Krypton (fixes issue
https://github.com/basilfx/plugin.audio.subsonic/issues/3)

### CODE REVIEW

File: addon.py [hunk 1]
```diff
@@ -31,8 +31,9 @@ def __init__(self, addon_url, addon_handle, addon_args):
         self.username = addon.getSetting("username")
         self.password = addon.getSetting("password")
 
-        self.random_count = addon.getSetting("random_count")
-        self.bitrate = addon.getSetting("bitrate")
+        self.random_count = int(addon.getSetting("random_count"))
```
Comment (basilfx) [inline addon.py]: Not sure what will happen if `addon.getSetting("random_count")` returns a non-integer. Maybe wrap it in something like

``` python
try:
    self.bitrate = int(addon.getSetting("random_count"))
except ValueError:
    self.bitrate = 0
```

Same applies for other calls below.

File: addon.py [hunk 2]
```diff
@@ -223,17 +211,47 @@ def genre_list(self):
...
+    def newest_albums_list(self):
+        """
+        Display newest album list.
+        """
+
+        size = self.album_list_size
+
+        xbmcplugin.setContent(self.addon_handle, "albums")
+
+        for album in self.connection.walk_album_list('newest',size,None,None,None):
```
Comment (basilfx) [inline addon.py]: PEP8: spaces betweeen arguments.

File: addon.py [hunk 3]
```diff
@@ -223,17 +211,47 @@ def genre_list(self):
...
+        """
+        Display album list by genre menu.
+        """
+
+        size = self.album_list_size
+        genre = self.addon_args["foldername"][0].decode("utf-8")
+
+        xbmcplugin.setContent(self.addon_handle, "albums")
+
+        for album in self.connection.walk_album_list('byGenre',size,None,None,genre):
```
Comment (basilfx) [inline addon.py]: PEP8: spaces betweeen arguments.

File: addon.py [hunk 4]
```diff
@@ -223,17 +211,47 @@ def genre_list(self):
...
         Display album list by genre menu.
         """
 
+        size = self.album_list_size
         genre = self.addon_args["foldername"][0].decode("utf-8")
 
         xbmcplugin.setContent(self.addon_handle, "albums")
 
-        for album in self.connection.walk_album_list_genre(genre):
+        for album in self.connection.walk_album_list('byGenre',size,None,None,genre):
```
Comment (basilfx) [inline addon.py]: PEP8: spaces betweeen arguments.

File: lib/libsonic_extra/__init__.py [hunk 5]
```diff
@@ -391,24 +391,30 @@ def walk_genres(self):
         for genre in response["genres"]["genre"]:
             yield genre
 
-    def walk_album_list_genre(self, genre):
+    def walk_album_list(self, ltype, size=10, fromYear=None,toYear=None, genre=None):
```
Comment (basilfx) [inline lib/libsonic_extra/__init__.py]: PEP8: spaces betweeen arguments.

File: addon.py (hunk 2 above)
Comment (basilfx) [reply addon.py]: I think you can also write `self.connection.walk_album_list('newest', size)`. The `None` arguments are kwargs.

File: addon.py (hunk 4 above)
Comment (basilfx) [reply addon.py]: I think you can also write `self.connection.walk_album_list('newest', size, genre=genre)`. The `None` are optional kwargs (they have defaults).

File: addon.py (hunk 3 above)
Comment (basilfx) [reply addon.py]: I think you can also write `self.connection.walk_album_list('newest', size, genre=genre)`. The None arguments are optional.

**Review by basilfx — CHANGES_REQUESTED** [review]
Thanks for this PR. If you address the changes, I am happy to merge them.

File: lib/libsonic_extra/__init__.py (hunk 5 above)
Comment (basilfx) [reply lib/libsonic_extra/__init__.py]: It is not Pythonic to have `camelCase` names. use `snake_case`

**Comment by gordielachance** [comment]
Hi, I made your changes...

I also reorganized the menus and added submenus for albums (Newest / Most played / Rencently played).

The one thing i'm not able to do (that's my first time with Python) is to limit the albums query to a 'size' (for pagination).  I don't understand why.
I did create a new option _albums_per_page_, and it seems that it is called correctly, but each time the whole list of albums is fetched; which is quite slow as I have hundreds of albums on Subsonic.

Could you have a look ?
I would be glad if you could review and correct my code; as I said I'm a total newbie in Python and this took me the whole day :)  Thanks !!!

**Comment by gordielachance** [comment]
Hi, I made a lot of changes in my fork !
Have a look if you have some time... There is now the ability to download and even star tracks :)

**Comment by gordielachance** [comment]
Hi basilfx,
Sorry to ask, but do you plan to merge my changes ?
I ask because if you prefer rather not, I could rename/split "my" version of the plugin so I could try to make it available on the official Kodi repositories.
Now that I spent so much time working on it (and thanks for your first version which was a great base); I would be glad if i'm not the only one to benefit from it :)
Thanks !

**Comment by basilfx** [comment]
I don't mind if you fork and work on your own version. That's how GitHub works :-)

After all, I don't use this plugin anymore, so as long as you respect the current [LICENSE](https://github.com/basilfx/plugin.audio.subsonic/blob/master/LICENSE) (don't forget to add yourself!) then I don't mind.

I'll even put deprecated notice in the README and redirect people to your fork.

**Pull request closed without merging**